### ISSUE TITLE
more than one consumer in the same group receive the message?

### ISSUE THREAD

**Issue by jinyuanxie** [spicavigo/kafka-rust#159]
I create a topic with 1 partition, and use the following code to create some consumers:

``` rust
Consumer::from_hosts(brokers)
    .with_topic("test".to_owned())
    .with_group("group-recognizer".to_owned())
    .with_fallback_offset(FetchOffset::Latest)
    .create()
    .unwrap()
```

then use the following code to listen the topic's message:

``` rust
for ms in consumer.poll().unwrap().iter() {
    for m in ms.messages() {
        println!("{:?}", m);
    }
    consumer.consume_messageset(ms);
}
consumer.commit_consumed().unwrap();
```

Also, I notice that **no output** like the following when those consumers start:
```
[2017-06-21 00:06:55,868] INFO [GroupCoordinator 0]: Assignment received from leader for group group-recognizer for generation 4 (kafka.coordinator.GroupCoordinator)
```

which will be generated when using `kafka-console-consumer.sh` with the same `group.id` in properties file.

Edit:
When add `.with_offset_storage(GroupOffsetStorage::Kafka)` to Consumer builder, the program panic on when **receiving the second** message:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Kafka(UnknownMemberId)', src/libcore/result.rs:860
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by dead10ck** [comment]
I'm sorry, I'm not understanding what the problem is from your description.

**Comment by dead10ck** [comment]
Also, can you tell me anything about what version your Kafka cluster is, and what kind of environment it's running in?

**Comment by jinyuanxie** [comment]
cc @dead10ck 
In a nutshell, when using the `kafka-console-consumer.sh` with `group.id` property, only **one consumer** instance will receive the message, but **all my consumer** instance will receive the message, which is not excepted.

* Kafka version: 0.9.0.1
 * OS: Ubuntu 16.04.1 LTS (GNU/Linux 4.4.0-53-generic x86_64)

Kafka topic create by:
`./kafka-topics.sh --create --zookeeper localhost:12181,localhost:12182,localhost:12183 --replication-factor 1 --partitions 1 --topic test`

client environment:
* OS: macOS 10.12.5
* `kafka-rust` version: 0.6.1
* `rust` version: rustc 1.19.0-nightly

**Comment by dead10ck** [comment]
@jinyuanxie oh, I think I understand. From what I gather, you're talking about how the official Java clients [handle multiple consumers on a single group ID](https://cwiki.apache.org/confluence/display/KAFKA/Consumer+Group+Example), yes?

The situation for this is unfortunate. If you look at the [Kafka protocol documentation](https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-FetchAPI) (particularly, the Fetch API), neither the Kafka protocol nor the server handle partition balancing at all.

> One thing to note is that the fetch API requires specifying the partition to consume from. The question is how should a consumer know what partitions to consume from? In particular how can you balance the partitions over a set of consumers acting as a group so that each consumer gets a subset of partitions. We have done this assignment dynamically using zookeeper for the scala and java client. The downside of this approach is that it requires a fairly fat client and a zookeeper connection. We haven't yet created a Kafka API to allow this functionality to be moved to the server side and accessed more conveniently. A simple consumer client can be implemented by simply requiring that the partitions be specified in config, though this will not allow dynamic reassignment of partitions should that consumer fail. We hope to address this gap in the next major release.

So basically, this feature of dynamically balancing the partitions across consumers is done by the official clients through Zookeeper, and from what I have been able to find, its usage of Zookeeper is not documented, so short of reverse engineering, we would not be able to handle dynamic partition reassignment in this lib.

So unfortunately, unless there is some documentation somewhere that I have not been able to find, your options are limited to waiting for Kafka to address this in a future version, or manually assigning partitions to distribute work among multiple consumers.

**Comment by dead10ck** [comment]
It's also worth noting that in your particular example, the [official documentation](https://cwiki.apache.org/confluence/display/KAFKA/Consumer+Group+Example) for the high level consumer mentions that

> if you provide more threads than there are partitions on the topic, some threads will never see a message

So your second consumer should actually never receive any messages.

**Comment by jinyuanxie** [comment]
Thanks for your reply, the dynamic balancing is handled by the client not 
 kafka server, right?

I think I know how to deal with my problem temporarily.

ps: Maybe I can make a discovery in `libkafka` to find out how to implement this feature after finish my assignment and final exam :(

Thank for your help.

**Comment by dead10ck** [comment]
Sorry about this. I think I'll close this issue until something changes though.

**Comment by Dzordzu** [comment]
Hi, I've been struggling the same issue. Can you explain to me, why even after commiting the message by one consumer, the other one is able to receive messages?
